app/uploaders/article_image_uploader.rb

```diff
@@ -0,0 +1,53 @@
+class ArticleImageUploader < CarrierWave::Uploader::Base
+  include CarrierWave::MiniMagick
+
+  def store_dir
+    # This will be the directory path in the CMS
+    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
```

Comment:
I definitely can, I just need to understand the publication flow a little more.

My current understanding is:

the `YYYY/MM/DD` path isn't set until the article is published, but the image needs to be uploaded while the article is still in `draft`

If the article is published already, this gives the right path:
```ruby
article.path.gsub(article.slug,'')
```

but if it isn't published yet, I am not sure how to *know* the path to save the image in